File: backend/app/models/company.py
"""
Company data models for API responses.
"""

from pydantic import BaseModel


class Company(BaseModel):
    """Company information model."""

    id: str
    name: str
    display_name: str
    color: str  # Hex color for UI theming
    logo_url: str  # URL path to company logo
    tenets_file: str  # Filename of company tenets document
    description: str  # Brief description of company evaluation criteria


class CompaniesResponse(BaseModel):
    """Response model for GET /api/companies endpoint."""

    companies: list[Company]

File: frontend/src/types/company.ts
/**
 * Company data types
 */

export interface Company {
  id: string;
  name: string;
  display_name: string;
  color: string;
  logo_url: string;
  tenets_file: string;
  description: string;
}

export interface CompaniesResponse {
  companies: Company[];
}
